rust_src/src/data.rs

```diff
@@ -789,4 +791,38 @@ pub fn logcount(value: EmacsInt) -> i32 {
     unsafe { rust_count_one_bits(value as usize) }
 }
 
+/// Set SYMBOL's function definition to DEFINITION, and return DEFINITION.
+#[lisp_fn]
+pub fn fset(mut symbol: LispSymbolRef, definition: LispObject) -> LispObject {
+    /* Perhaps not quite the right error signal, but seems good enough.  */
+    let sym_obj = symbol.as_lisp_obj();
```

Comment:
Does leaving `symbol` a `LispObject` help?

(side note. We do not use C style comments. Replace /* */ with //.)